lib/v2/datasource/remote/api/rates_repository.dart

```diff
@@ -16,4 +16,27 @@ class RatesRepository extends NetworkRepository {
             }))
         .catchError((error) => mapHttpError(error));
   }
+  
+  // TODO-NIK code below needs to be put in a future with a wait all
```

Comment:
Look into `GetExploreUseCase` to learn how to make multiple calls in parallel. 
I could also show you the new app architecture anytime this week.

Reply:
Yes let's do that, this is a perfect case I need to make work, perfect to learn and see how we can do what we need to do inside the architecture.

In this case, there's a lot of questions

- Parallel calls - GetExploreUseCase shows how to combine futures so will be something like that
- Caching results - we don't want to call this more often than a certain threshold, say 1 time / hour
- Converting results into the object - we want to cache the objects, not the raw JSON of the results - not sure where inside the structure we do that. This was inside the notifiers before.
- This fiat rate result here is used in the list of currencies, but we also need to use it for actual conversion. Maybe that's not yet switched over, which would make sense (only new screens have the new architecture)

Reply:
Well these are my suggestions.

- Parallel calls - **Yes**

- Caching results - **Reviewing the original code, I see that to use that logic, a service for rates needs to be created that remains alive during the use of the app (such as ratesNofitier), We can create a Bloc for the rates that also would be globally accessible to the app .**

- Converting results into the object - we want to cache the objects, not the raw JSON of the results - **The previous answer would solve it, we could even keep these states in the cache of the app once it starts using the HydratedBloc lib but this is an extra and I'm not sure if that is what we want.**

- This fiat rate result here is used in the list of currencies, but we also need to use it for actual conversion. Maybe that's not yet switched over, which would make sense (only new screens have the new architecture) - 
**Yes, in fact only when something is required is passed to V2.**

Finally it is good to let us know @n13 of these considerations many times at least speaking for me to overlook some important characteristics and behaviors.